File: src/lib/scala/Slate.Common/src/main/scala/slate/common/Content.scala
/**
<slate_header>
  url: www.slatekit.com
  git: www.github.com/code-helix/slatekit
  org: www.codehelix.co
  author: Kishore Reddy
  copyright: 2016 CodeHelix Solutions Inc.
  license: refer to website and/or github
  about: A Scala utility library, tool-kit and server backend.
  mantra: Simplicity above all else
</slate_header>
  */

package slate.common

/**
 * represents a container for content and the conetnt type.
 * e.g.
 *
 * 1. string, json
 * 2. string, csv
 *
 * @param text
 * @param format
 */
case class Content(text:Option[String], format:String) {

  /**
   * whether this content is empty
   * @return
   */
  def isEmpty: Boolean = text.isEmpty


  /**
   * whether this content is present
   * @return
   */
  def isDefined: Boolean = text.isDefined


  /**
   * the length of the content
   * @return
   */
  def size : Int = text.fold(0)( t => t.length )


  /**
   * extension of the format
   * @return
   */
  def extension: String = format
}


object Content {

  def csv(text:String):Content = new Content(Option(text), "csv")


  def json(text:String):Content = new Content(Option(text), "json")


  def props(text:String):Content = new Content(Option(text), "props")
}

File: src/lib/kotlin/slatekit-core/src/main/kotlin/slatekit/core/push/Content.kt
/**
<slate_header>
url: www.slatekit.com
git: www.github.com/code-helix/slatekit
org: www.codehelix.co
author: Kishore Reddy
copyright: 2016 CodeHelix Solutions Inc.
license: refer to website and/or github
about: A Kotlin utility library, tool-kit and server backend.
mantra: Simplicity above all else
</slate_header>
 */
package slatekit.core.push

import slatekit.common.DateTime


/**
 * The content is a fixed set of metadata specifically about the payload
 * and is actually separately stored and outside of the dynamic payload structure
 * This can be used for fast lookup / hints of the actual payload content
 */
data class Content(
    val title: String,
    val desc: String,
    val date: DateTime? = null,
    val args: Map<String,String>? = null
) {

    companion object {
        val empty = Content("", "")
    }
}
